docker/CMSRucioClient/traces/collect_send_traces.py

```diff
@@ -0,0 +1,160 @@
+    We set the search time interval from "now-12m" to now for  a 12 min window. We plan to run the cron for every 10 minutes.   
+    """
+
+    print ("**** Starting time **** ", time.asctime(time.gmtime()))
+    t1 = int(time.time())
+
+    QUERY_HEADER = '{"search_type":"query_then_fetch","ignore_unavailable":true}'
+
+    with open('query_collect.json', 'r') as WMArchive_json:
+        wma = json.load(WMArchive_json)
+
+    # metadata.timestamp uses ms for unit.      
+    trange = {
+            "range": {
+                        "metadata.timestamp":{
+                                            "gte": int(time.time() - 12*60)*1000,
+                                            "lte": int(time.time())*1000
+                                            }
+                        }
+            }
+    # adding time
+    wma["query"]["bool"]["must"].append(trange)
+    print("**** ES Query ****  ", json.dumps(wma))
+    query = io.StringIO(QUERY_HEADER + '\n' + json.dumps(wma) + '\n')
+    
+    headers = {'Authorization': 'Bearer %s' % os.environ['MONIT_TOKEN'],
+            'Content-Type': 'application/json'}
+    
+    r = requests.post('https://monit-grafana.cern.ch/api/datasources/proxy/9545/_msearch', data=query, headers=headers)
+    if not (200 <= r.status_code <= 229):
+        print("Error for query ES. Http error code: ", r.status_code)
+        print(r.text)
+        #TODO we need wait and redo. How many times to try? In addition, we need to have better error handing when running with cron
+        exit(1)
+    #
+    result = json.loads(r.text)
+    t2 = int(time.time())
+    print("**** Time spend on ES **** ", t2-t1)
+    traces = []
+    moreQuery = False
+    hits = 0 
+    for key, value in result.items():
+        #print(json.dumps(value[0]))
+        for v in value:
+            #there are two ways to get the total hits and they shoube the same
+            l = len(v['hits']['hits'])
+            total = v['hits']['total']
+            if ( l != total):
+                #TODO raise exception?
+                print ("Error: result array size ", l , "is not eaual to total hits ",  total)
+                exit(1)
+            print("*** total number of hits: ", total)
+            if total > 10000:
+                moreQuery = True
+                hits += 1
+            
+            for h in v['hits']['hits']:
+                data = h['_source']['data']
+                LFNArrayRef = data['LFNArrayRef']
+                fallbackFiles = data['fallbackFiles']
+                LFNArray = data['LFNArray']
+                
+                trace = {}
+                goodlfn = []
+                site = ''
+                ts = data['meta_data'].get('ts', int(time.time()))
+                jobtype = data['meta_data'].get('jobtype', 'unknown')
+                wn_name = data['meta_data'].get('wn_name', 'unknown')
+
+                for s in data['steps']:
```

Comment:
Generally I'd rather see "step" than "s" here and it's especially nice for my next comment.